modules/openapi-generator/src/main/resources/dart/libraries/dio/serialization/json_serializable/deserialize.mustache

```diff
@@ -60,5 +60,5 @@ final _regMap = RegExp(r'^Map<String,(.*)>$');
           }
           break;
     } 
-    throw Exception('Cannot deserialize');
```

Comment:
This looks wrong, do you have a stacktrace for the error?
```dart
abstract interface class Exception {
  factory Exception([var message]) => _Exception(message);
}
```

Reply:
it's a compiler error for when using json serializer with these arguments

```
-g dart-dio --additional-properties=serializationLibrary=json_serializable
```

this is the generated exception

```
@JsonSerializable(
  checked: true,
  createToJson: true,
  disallowUnrecognizedKeys: false,
  explicitToJson: true,
)
class Exception {
  /// Returns a new [Exception] instance.
  Exception({

     this.targetSite,

     this.message,

     this.data,

     this.innerException,

     this.helpLink,

     this.source_,

     this.hResult,

     this.stackTrace,
  });
```

Reply:
You seem to have a custom `Exception` class/element in your schema that conflicts with the default `Exception` class from Dart.
Please rename the element in your schema or follow https://openapi-generator.tech/docs/customization/#name-mapping to map it if you can't change your schema.

Reply:
interesting, the swashbuckle .net generator does indeed create an Exception definition. 

Perhaps the OpenApi exception should be renamed instead, as this seems to be picking up the C# Exception implementation. I'm suprised im the first one to run into this issue.

either way, the change isn't valid you're correct. closing.

Reply:
This is the Dart `Exception` class, we have no way of renaming it.

Reply:
I was thinking about using the ApiException, since that's also used throughout the code, except for this instance.